### PR TITLE
Replication performance improvements

### PR DESCRIPTION
**Pull request by groldan**
PR to review and discuss the performance improvements for clone/fetch/push.

### CODE REVIEW

File: src/storage/rocksdb/src/main/java/org/locationtech/geogig/rocksdb/RocksConnectionManager.java [hunk 1]
```diff
@@ -178,6 +181,13 @@ private ColumnFamilyDescriptor newColDescriptor(String name) {
 
     private ColumnFamilyOptions newColFamilyOptions() {
         ColumnFamilyOptions colFamilyOptions = new ColumnFamilyOptions();
+
+        // enable bloom filter to speed up RocksDB.get() calls
+        BlockBasedTableConfig tableFormatConfig = new BlockBasedTableConfig();
```
Comment (DBlasby) [inline src/storage/rocksdb/src/main/java/org/locationtech/geogig/rocksdb/RocksConnectionManager.java]: This probably should have come in as a separate PR -- but its fine.

File: src/storage/rocksdb/src/main/java/org/locationtech/geogig/rocksdb/RocksdbObjectStore.java [hunk 2]
```diff
@@ -419,7 +419,7 @@ public final void putAll(Iterator<? extends RevObject> objects, final BulkOpList
     protected void putAll(Stream<RevObject> stream, BulkOpListener listener) {
         final Stopwatch sw = LOG.isTraceEnabled() ? Stopwatch.createStarted() : null;
 
-        final int batchsize = 10_000;
+        final int batchsize = 1000;
```
Comment (DBlasby) [inline src/storage/rocksdb/src/main/java/org/locationtech/geogig/rocksdb/RocksdbObjectStore.java]: did we do any performance testing with this.  I think there's some matching # in the GeoTools/GeoServer feature iterators....

File: src/storage/rocksdb/src/main/java/org/locationtech/geogig/rocksdb/RocksdbObjectStore.java [hunk 3]
```diff
@@ -443,7 +443,8 @@ private int insertBatch(Iterator<EncodedObject> objects, BulkOpListener listener
         try (RocksDBReference dbRef = dbhandle.getReference();
                 WriteOptions wo = new WriteOptions(); //
                 WriteBatch batch = new WriteBatch()) {
-            wo.setSync(true);
```
Comment (DBlasby) [inline src/storage/rocksdb/src/main/java/org/locationtech/geogig/rocksdb/RocksdbObjectStore.java]: does this need to be un-done or are these changes just for the this write op?

File: src/core/src/test/java/org/locationtech/geogig/plumbing/diff/DiffCountConsumerTest.java [hunk 4]
```diff
@@ -82,7 +82,7 @@ private void createFeatureTypesTree(RevTreeBuilder rootBuilder, String treePath,
     }
 
     private DiffObjectCount count(RevTree left, RevTree right) {
-        PreOrderDiffWalk visitor = new PreOrderDiffWalk(left, right, odb, odb);
+        PreOrderDiffWalk visitor = new PreOrderDiffWalk(left, right, odb, odb, true);
```
[the remark this answers is not in the document]
Comment (groldan) [reply src/core/src/test/java/org/locationtech/geogig/plumbing/diff/DiffCountConsumerTest.java]: no, it's not, it's a leftover from an experiment, good catch

File: src/core/src/main/java/org/locationtech/geogig/plumbing/UpdateRef.java [hunk 5]
```diff
@@ -52,8 +56,8 @@ public UpdateRef setName(String name) {
      *        {@code "refs/origin/master"}
      * @return {@code this}
      */
-    public UpdateRef setNewValue(ObjectId newValue) {
-        this.newValue = newValue;
+    public UpdateRef setNewValue(@Nullable ObjectId newValue) {
+        this.newValue = newValue == null ? null : newValue;
```
[the remark this answers is not in the document]
Comment (groldan) [reply src/core/src/main/java/org/locationtech/geogig/plumbing/UpdateRef.java]: ha!

File: src/remoting/NOTICE.txt [hunk 6]
```diff
@@ -0,0 +1,4 @@
+The Content includes source code from the following projects:
+
+* Varint from the Apache Mahout under Apache License Version 2.0
```
[the remark this answers is not in the document]
Comment (groldan) [reply src/remoting/NOTICE.txt]: ha, no, the file got copied over from geogig-core but it's not needed, will delete it.

File: src/remoting/src/main/java/org/locationtech/geogig/remotes/CloneOp.java [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+                final Integer depth = this.depth.or(remoteRepo.getDepth()).or(0);
+                setDepth(cloneRepo, depth);
+                localRemoteRefs = fetchRemoteData(cloneRepo, remote, depth);
+            }
+            setUpRemoteTrackingBranches(cloneRepo, remote, localRemoteRefs);
+        } catch (Exception e) {
+            // something went wrong, need to delete the clone
+            cloneRepo.close();
+
+            // TODO: enable when this commands actually creates the
```
[the remark this answers is not in the document]
Comment (groldan) [reply src/remoting/src/main/java/org/locationtech/geogig/remotes/CloneOp.java]: it currently is by CLI's clone, but it should be CloneOp's responsibility as well as actually creating the target repo. To be done later though, once we can create `memory://` repositories for testing

File: src/storage/rocksdb/src/main/java/org/locationtech/geogig/rocksdb/RocksdbObjectStore.java (hunk 2 above)
Comment (groldan) [reply src/storage/rocksdb/src/main/java/org/locationtech/geogig/rocksdb/RocksdbObjectStore.java]: I did empitically, although I've work in progress in the `geogig-benchmarks` project to do proper rocks backend performance testing

File: src/storage/rocksdb/src/main/java/org/locationtech/geogig/rocksdb/RocksConnectionManager.java (hunk 1 above)
Comment (groldan) [reply src/storage/rocksdb/src/main/java/org/locationtech/geogig/rocksdb/RocksConnectionManager.java]: yeah

**Pull request merged**